server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -2584,6 +2585,54 @@ protected void runInContext() {
         }
     }
 
+    private void verifyVmLimits(UserVmVO vmInstance, Map<String, String> details) {
+        if (VirtualMachineManager.ResourceCountRunningVMsonly.value()) {
```

Comment:
@Pearl1594 
I think we need to move line 2589-2591 to line 2609.

Reply:
Just to confirm before making the change - @weizhouapache this is to ensure that before updating the VM settings for cpuNumber/ memory - we validate it is within the limits of the custom offering it uses (if it uses)?

Reply:
@Pearl1594 yes, exactly.

otherwise, if ResourceCountRunningVMsonly is true, the validation of customer offering is not executed.

Reply:
Yeah - makes sense, I'll update it.